### PR TITLE
Add OpenLDAP secrets engine demonstration

### PR DESCRIPTION
**Pull request by brianshumate**
The goal here is to deconstruct the [original OpenLDAP + sshd PoC](https://github.com/hashicorp/vault-tools/tree/master/users/jodonnell/openldap-ssh-poc) into a step-wise guide that can serve as a deeper dive bonus round to the main OpenLDAP secrets engine guide (which tries to be both in depth and generically useful) that is currently in review here:

https://github.com/hashicorp/learn/pull/1165.

This also extends the original PoC to include a Vault container and works through an example of credential rotation and read.

It will be linked from the main guide as an optional continuation once it is merged here in vault-guides.

### CODE REVIEW

**Comment by burtlo** [comment]
I would consider dropping the `demo` from the directory name.

File: secrets/openldap/Dockerfile.centos7 [hunk 1]
```diff
@@ -0,0 +1,26 @@
+FROM jdeathe/centos-ssh:2.6.1
```
Comment (jasonodonnell) [inline secrets/openldap/Dockerfile.centos7]: I wonder if we should just publish the docker image on dockerhub instead of having folks build it?

Comment (brianshumate) [reply secrets/openldap/Dockerfile.centos7]: Sounds like a great idea especially towards future promotion and reuse!

Comment (brianshumate) [reply secrets/openldap/Dockerfile.centos7]: I am not sure who has the ability to publish such a container image, but I don't think that I do.

Is it okay to use this as-is for now with a goal to eventually build a docker image going forward?

**Pull request merged**